### DISCUSSION TITLE
can i benchmark with redis-benchmark ?

### DISCUSSION THREAD

**Discussion by osevan** [danielealbano/cachegrand#240] [Q&A] [answered]
I hope i can benchmark with onboard userspace tool redis-benchmark.

It is possible?

**Comment by danielealbano** [comment, marked as answer]
You should be able to but redis-benchmark doesn't scale up as memtier_benchmark does so you might not be vale to see a great improvement over Redis or other platforms.
Even with memtier_benchmark i have encountered quite some performance limitations and had to scale up to two different machines to actually saturate one.

Also, Redis supports more than 300 commands, cachegrand much less, so you might want to be specific in the tests you run.

Said that, we are working on leveraging the unit tests to automatically generate benchmarks and the cool thing is that we will be able to cover all the scenarios where the commands can perform well (or less well) and will be possible to run these tests also on different platforms.

@Valkyrie00 this is related to your recent efforts!

**Reply by danielealbano** [reply]
Just in a side note, cachegrand does position itself as well as Redis replacement, or to be more precise as Redis Enterprise replacement, with a few more goals and open source :)

As this is a full open source project, non sponsored by any company on investor, and we are working on it during our fee time, it's going to take a bit more time.
